functions/test/endpoints/FirestoreDataFactory.ts

```diff
@@ -333,6 +333,35 @@ export class FirestoreDataFactory{
         })
     }
 
+    /**
+     * Create an event
+     * @param db 
+     * @param creator_id
+     * @param eOpts 
+     */
+    static setEvent(db: firebase.firestore.Firestore, id: string, creator_id: string, eOpts:Options.EventOptions = Options.EVENT_DEFAULTS): Promise<void> {
+        const data = {
+            "Name": (eOpts.name !== undefined)?eOpts.name:Options.EVENT_DEFAULTS.name,
+            "Details": (eOpts.details !== undefined)?eOpts.details:Options.EVENT_DEFAULTS.details,
+            "Date": (eOpts.date !== undefined)?eOpts.date:Options.EVENT_DEFAULTS.date,
+            "Location": (eOpts.location !== undefined)?eOpts.location:Options.EVENT_DEFAULTS.location,
+            "Points": (eOpts.points !== undefined)?eOpts.points:Options.EVENT_DEFAULTS.points,
+            "PointTypeID": (eOpts.point_type_id !== undefined)?eOpts.point_type_id:Options.EVENT_DEFAULTS.point_type_id,
+            "PointTypeName": (eOpts.point_type_name !== undefined)?eOpts.point_type_name:Options.EVENT_DEFAULTS.point_type_name,
+            "PointTypeDescription": (eOpts.point_type_description !== undefined)?eOpts.point_type_description:Options.EVENT_DEFAULTS.point_type_description,
+            "House": (eOpts.house !== undefined)?eOpts.house:Options.EVENT_DEFAULTS.house,
+            "CreatorID":creator_id
+        }
+        return db.collection("Events").doc(id).set(data)
+    }
+
+    /**
+     * Run this after events test to clean events
+     * @param db Test App Firestore instance
+     */
+    static async cleanEvents(db: firebase.firestore.Firestore) {
+        await FirestoreDataFactory.deleteCollection(db, "Events", 1)
```

Comment:
The last parameter is the number of items to remove in a batch delete, so if you increase the number to 100 it decreases the overhead of the delete requests.